pkg/checker/checker.go

```diff
@@ -6,49 +6,113 @@ package checker
 
 import "fmt"
 
-// Checker is the type of checking functions
-type Checker func() error
+// ResultOK is the string to appear in CheckResult.Result if the check was successful
+const ResultOK = "OK"
 
-// Remediator is the type of remediation functions
-type Remediator func() error
+//ResultError is the string to appear in CheckResult.Result if the check failed
+const ResultError = "ERROR"
 
-// Check is a type that implements a netboot check
+// CheckFun is the type of checking functions
+type CheckFun func(CheckArgs) error
+
+// CheckArgs is the argument map to be passed to CheckFuns
+type CheckArgs map[string]string
+
+// A Check describes a call to a CheckFun with possible remediations
```

Comment:
So.... what is a Check? Or a Checker? Most of these have really generic godocs, which is not helpful.

What does it check? Why? What is this for?

Reply:
You are right, I'll look into documenting this better.